### ISSUE TITLE
error on DESI to Photo-z workflow: " Filter DESI/bass.g missing from the model grid or not in the right order "

### ISSUE THREAD

**Issue by Andrei-EPFL** [esg-epfl-apc/tools-astro#165]
I have made a workflow with just these 2 tools:
https://galaxy.odahub.fr/u/andreiv/w/copy-of-atel-follow-up-photometric-data-legacy-survey

If I put in the dictionary of filters the DESI filters first and then the DECam ones, I get an error:
"2024-12-17T16:44:35UTC ElementsProgram FATAL : # Elements Exception : Filter DESI/bass.g missing from the model grid or not in the right order" 
i.e. when I use this dictionary:
[Galaxy136-[DESI Legacy Survey (PR 146) -_ Catalog dictionary_filters].json](https://github.com/user-attachments/files/18169120/Galaxy136-.DESI.Legacy.Survey.PR.146.-_.Catalog.dictionary_filters.json)

If I put the DECam filters first, the workflow on galaxy works fine.

1) On renkulab of Photo-z Euclid, I cannot reproduce this issue.
2) The workflow on galaxy has now DESI then DECam filters. So theoretically you could reproduce this error by putting:
Source name =  Mrk421 
Ra  = 166
Dec = 38  

Thanks !

**Comment by Andrei-EPFL** [comment]
I forgot to mention that Marco said that the order should not matter.

**Comment by Andrei-EPFL** [comment]
I have done more comparisons and I observed that phosphorus on galaxy has version 2.0.6 and on renkulab (the one I use) has 2.0.0.  
I am trying now to install 2.0.6 on renkulab and do more tests, but it takes ages to install it. When I have more tests I will write them here.

**Comment by Andrei-EPFL** [comment]
@dsavchenko I have found the problem. The version of phosphoros is indeed the problem. v2.0.0 works fine. v2.0.6 provides this error. I forked photo-z-euclid and installed v2.0.6 and I managed to reproduce this error in renkulab.

Now, there is another thing related to phosphoroscore: https://github.com/astrorama/PhosphorosCore it seems that there would be a 2.1 but it is not anaconda yet. this may not be in our control.
Therefore, I think that if we fix the version to 2.0.0 may be fine for galaxy and renku, for a while. I will also open an issue on gitlab of photo-z to mention it to Marco.

**Comment by bgruening** [comment]
They seem to have their own channel: https://anaconda.org/astrorama/phosphoroscore/files

**Comment by Andrei-EPFL** [comment]
Thank you! I realized that I confused a bit phosphoroscore and phosphoros and their versions. so my second paragraph in my previous comment is wrong. 

Nevertheless the difference between phosphoros v2.0.0 and v2.0.6 stands.
I installed v2.0.6 on https://renkulab.io/projects/andrei.variu/photo-z-euclid-test and I can reproduce the error mentioned in this issue.

I have opened another issue on gitlab of the photo-z:https://gitlab.renkulab.io/astronomy/mmoda/photoz-euclid/-/issues/17  therefore I think I can close this issue. 
Thank you!!
